staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/pod.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceclaim
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodStatusDeepEqual is a manually written DeepEqual for
+// [corev1.PodResourceClaimStatus] slices.
+func PodStatusDeepEqual(statusA, statusB []corev1.PodResourceClaimStatus) bool {
+	if len(statusA) != len(statusB) {
+		return false
+	}
+	// In most cases, status entries only get added once and not modified.
+	// But this cannot be guaranteed, so for the sake of correctness in all
+	// cases this code here has to check.
+	for i := range statusA {
```

Comment:
Could statusA and statusB have the same items, but with different orders?

Reply:
In theory, yes. In practice that should never happen, so adding a `slices.Sort` here just adds performance overhead in the common case that doesn't pay off.